### PR TITLE
[3077] New issue "(push|emplace)_back should invalidate the end itera…

### PR DESCRIPTION
**Pull request by CaseyCarter**
…tor"

### CODE REVIEW

File: xml/issue3077.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+<sref ref="[container.requirements.general]"/> paragraph 12:
+<blockquote>
+Unless otherwise specified (either explicitly or by defining a function in terms
+of other functions), invoking a container member function or passing a container
+as an argument to a library function shall not invalidate iterators to, or change
+the values of, objects within that container.
+</blockquote>
+It seems that this second sentence (1) should be a note that reminds us that the
+blanket wording applies here when no reallocation occurs, and/or (2) actually
+intends to specify that iterators at and after the insertion point are invalidated.
```
Comment (timsong-cpp) [inline xml/issue3077.xml]: I think it's meant to do this and we should rephrase it accordingly rather than making it a note.

Comment (CaseyCarter) [reply xml/issue3077.xml]: The PR turns this sentence into a note to accomplish purpose two, and introduces new wording for purpose one. (I see no need to invalidate iterators *after* the end, which I believe cannot be valid on any case.)

Comment (timsong-cpp) [reply xml/issue3077.xml]: This sentence is in a paragraph that applies to all insertion operations, including `insert` and `emplace`, not just `push/emplace_back`. The first two can insert in the middle, in which case we need to invalidate everything at or after the insertion point.

Comment (CaseyCarter) [reply xml/issue3077.xml]: Yes, thanks - I'll fix this.

**Pull request merged**